stroyprombeton/context.py

```diff
@@ -174,14 +146,65 @@ def context(self) -> typing.ContextDict:
             # @todo #431:60m  Move total_products to a relevant context class.
             #  At SE side "paginated" context var somehow contains "total_products" field.
             #  And move the relevant `test_tag_button_filter_products` test to the fast tests side.
-            'total_products': products.count(),
+            'total_products': options.qs().count(),
             **pages_context.Contexts([
-                page, category, sliced_products,
+                page, category, sliced_options,
                 images, grouped_tags
             ]).context()
         }
 
 
+class TagsFilter(context.Tags):
+
+    def __init__(
+        self,
+        tags: stb_models.TagQuerySet,
+        request_data_: request_data.FetchProducts
+    ):
+        super().__init__(qs=tags)
+        self.request_data = request_data_
+
+    def qs(self) -> stb_models.TagQuerySet:
+        selected_tags = context.tags.ParsedTags(
+            tags=context.Tags(self._qs),
+            raw_tags=self.request_data.tags,
+        )
+        if self.request_data.tags:
+            selected_tags = context.tags.Checked404Tags(selected_tags)
+        return selected_tags.qs()
+
+
+class OptionsFilter(context.Context):
+    def __init__(
+        self,
+        category: stb_models.Category,
+        tags: stb_models.TagQuerySet,
+        request_data_: request_data.FetchProducts
+    ):
+        self.category = category
+        self.tags = tags
+        self.request_data = request_data_
+
+    def qs(self) -> stb_models.OptionQuerySet:
+        return (
+            stb_models.Option.objects.active()
+            .bind_fields()
+            .active()
+            .filter_descendants(self.category)
+            .tagged_or_all(self.tags)
+            .order_by(*settings.OPTIONS_ORDERING)
+        )
+
+    def context(self) -> typing.ContextDict:
+        return {
+            'positions': self.qs()
+        }
+
+
+# @todo #449:60m  Remove FetchPositions class.
+#  Move the class's filter_positions and slice_products
+#  to a new related context classes.
```

Comment:
```suggestion
#  to new related context classes.
```

Reply:
What classes do you mean? Should implementer to create them or do they exist?

Reply:
i meant created ones. I'll fix the comment